test.c

```diff
@@ -2476,7 +2637,9 @@ message_eq (int index, int connect, const struct message *expected)
     if (!r) return 0;
   }
 
-  MESSAGE_CHECK_STR_EQ(expected, m, upgrade);
+  if (!connect) {
+    MESSAGE_CHECK_STR_EQ(expected, m, upgrade);
+  }
```

Comment:
Test runner bug fix.  This check doesn't make sense for `test_message_connect(&responses[i])` (which is what calls this function) but it went unnoticed because none of the responses had the `.upgrade` field set.